### PR TITLE
Add changelog

### LINKED ISSUE #406: FR: Changelog
A changelog in the app might be a good point to notify the user about new features and fixes.

### PR DESCRIPTION
**Pull request by mueller-ma**
Fixes #406

### CODE REVIEW

**Comment by mueller-ma** [comment]
I'm not sure if thats the best format for changelog

**Comment by digitaldan** [comment]
So my only thing about a change log like this is it has to be manually updated, it seems like we have a canonical source of changes, github, can we do something with tagging on PR's and issues and simply have a search that represents all changes (so a search for everything tagged 2.0). ?  WDYT?

**Comment by ThomDietrich** [comment]
@mueller-ma could be an interesting read: http://keepachangelog.com/en/1.0.0/

**Comment by mueller-ma** [comment]
There are tool to generate changelogs, but most PRs are untagged. I will look at some tools and start tagging PRs for the next release

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+## Additions
+- Add adaptive app icon
+- Add two new themes
+- Add app intro
+- Add about screen
+- Implement support for HSL videos
+- Add theme support for charts. Need openHAB server > 2.1
+
+## Improvements
```
Comment (lolodomo) [inline CHANGELOG.md]: Please add as a new feature "add support for image items".

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- Update app icon
+- Improved selection and setpoint items
+- Improved error messages
+- Improved charts size. Need openHAB server > 2.1
+- Better support for image items
```
Comment (lolodomo) [inline CHANGELOG.md]: Maybe something like "better handling of images (refresh, scaling, ...)".

File: CHANGELOG.md (hunk 1 above)
Comment (mueller-ma) [reply CHANGELOG.md]: This might be misleading. Image items worked in the past.

Comment (lolodomo) [reply CHANGELOG.md]: Only image through URL was implemented, not image items.

**Comment by digitaldan** [comment]
LGTM

**Comment by digitaldan** [comment]
Lets leave this open until we have all the other 2.0  PR's merged, just in case we need to add something.

**Comment by FlorianSW** [comment]
A general suggestion: Having the github issue mentioned next to the fixes (and probably the feature improvements, too, if there were any) would be great, so anyone could follow why we did what and could see the linked PRs and so on.

Also: I would like to see the git sha1 mentioned, too, but that could be something I personally like, to directly see which commit is meant with this line in the changelog.

I also would like to have an automated process here, instead of needing someone to sit down and write this changelog. In another project I'm active we've a release-notes file which mentions the most important changes for the release we're currently working on. This file is managed by the persons, who made noteable changes, which means, that, e.g., if I change something important, in the same PR I change the release-notes file, too. This can, however, result in a lot of merge conflicts :/

While I understand, that a git log dump is probably not always the best idea, it's at least complete (read: it contains anything and no-one filters it :)) and automated. And git-log can do a lot of fancy stuff to have a useable and readable output, and we could also parse the git-log output ourselfs and do whatever we need to do to get a fancy output (that's what, e.g., Wikimedia does with the wmf-releases of MediaWiki, a weekly release which is deployed to the Wikipedia, which looks like https://www.mediawiki.org/wiki/MediaWiki_1.31/wmf.1).

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+- Add two new themes
+- Add app intro
+- Add about screen
+- Implement support for HSL videos
+- Add theme support for charts. Need openHAB server > 2.1
+
+## Changed
+- Update app icon
+- Improved selection and setpoint items
+- Move settings to the side menu
```
Comment (FlorianSW) [inline CHANGELOG.md]: Should mention #391, too, which is now merged :)

**Comment by mueller-ma** [comment]
I tried https://github.com/skywinder/github-changelog-generator and thats the result: https://gist.github.com/mueller-ma/8b343976c5f121e16ba2252ec7d820a6
When we (Dan and me) start tagging every issue and PR, things like minor bug fixes and invalid issued can be filtered out. So I think this one is something for 2.0.1.

@FlorianSW I think a git-log is too technical for places like the play store.

**Comment by mueller-ma** [comment]
I started tagging the open prs. We also need a tag that indicates "This issue shouldnt show up in the changelog since it fixes an issue, that wasnt present in the last stable version". @digitaldan any ideas how to name that tag?

**Comment by digitaldan** [comment]
>"This issue shouldnt show up in the changelog since it fixes an issue, that wasnt present in the last stable version".

"unreleased-bugfix" ?

**Comment by lolodomo** [comment]
Don't forget to mention #501.

**Comment by digitaldan** [comment]
For the 2.0 release we will also need a German version of these notes for the Play store, any volunteers?  (this I can not help with)

**Comment by mueller-ma** [comment]
@digitaldan I tagged all prs. Please have a look at my branch changelog2

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- Fixed overlapping label and value
+- Fixed a bug where images and charts were not shown
+- Fixed a bug where notification were not shown when on WIFI
+- Fixed a bug where no charts was shown when icon type set to SVG
+- Fixed some bugs where wrong icon was shown
+- Removed useless sitemap refreshes
+
+## Removed
+- Remove support for controlling items via volume buttons
+- Remove Bindings and Discovery funtionality. Please use PaperUI instead
```
Comment (digitaldan) [inline CHANGELOG.md]: "functionality"

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# 2.0
+
+## Added
+- Add adaptive app icon
+- Add two new themes
+- Add app intro
+- Add about screen
+- Implement support for HSL videos
+- Add theme support for charts. Need openHAB server > 2.1
+- Added seperate user and password settings for local and remote connection
```
Comment (digitaldan) [inline CHANGELOG.md]: "separate"

**Comment by FlorianSW** [comment]
@digitaldan Once the Changelog is finished ("ready for merge"), I can translate it to german, do you need it somehwere in this repository or where should I paste it? :)

**Comment by mueller-ma** [comment]
@FlorianSW I dont think that we need a german version of the changelog in this repo, just a short german version for the play store. https://github.com/openhab/openhab.android/issues/499#issuecomment-351458443

**Comment by digitaldan** [comment]
Under "changed" should we mention "Moved NFC menu entry to item long press menu"  or something similar?

**Comment by FlorianSW** [comment]
Absolutely! :)

**Comment by digitaldan** [comment]
I think once that is added we can merge this one too.

**Comment by mueller-ma** [comment]
@digitaldan As a maintainer you should have write access to the branch in my repo. Can you add the point?

**Pull request merged**